### ISSUE TITLE
fastai2 integration

### ISSUE THREAD

**Issue by dreamflasher** [wandb/client#1597]
**Describe the bug**
The integration with fastai was moved to fastai with fastai2. Nevertheless, I'd aprecciate if you could have a look at the implementation, because currently it's broken. You introduce a breaking change in your API by increasing the step counter with every log call (now one needs to explicitly turn `commit=False` off, in order to have the original behaviour). I tried to fix this over in fastai with the following PR: https://github.com/fastai/fastai/pull/3067

Yet, something is still off and I'd appreciate your help in debugging/fixing.

I train my model for 5 epochs, and this is the accuracy plot in wandb:
![image](https://user-images.githubusercontent.com/31695/101783119-27dd9480-3afa-11eb-8cbe-46f016e78291.png)
And this is the fastai training output:
```
epoch     train_loss  valid_loss  accuracy  b_score  ece       nll       err  time    
0         7.291822    6.423790    0.891155  0.022018     0.056384  6.423790  0.122854           05:19    
1         6.633485    6.526669    0.906586  0.019923     0.055146  6.526669  0.132975           05:17     
2         7.929371    7.037761    0.877101  0.028144     0.071046  7.037761  0.144133           05:17     
3         6.300512    6.006740    0.909066  0.017325     0.049415  6.006740  0.115119           05:17   
4         5.873019    5.817275    0.922017  0.015450     0.049534  5.817275  0.111537           05:17      
```
The wandb plot in numbers:
```
epoch 
0     0.8912
1     0.9066
2     0.8771
3     0.9091
4     0.01545
```

So something is broken in the last epoch. 
`wandb, version 0.10.11`

```
python3 -c "import fastai; import torch; import fastcore; print(torch.__version__, fastai.__version__, fastcore.__version__)"
1.7.0+cu110 2.1.8 1.3.9
```

**Comment by borisdayma** [comment]
Hi @dreamflasher ,

Do you have any reproducible code you could share so that we can see the issue?
Providing the step explicitly ensures we associate validation and training data to the correct step.

**Comment by dreamflasher** [comment]
Let's close this issue in favor of the one over at fastai.
